app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+    // If the replica set has pods, show it.
+    if (_.get(replicaSet, 'status.replicas')) {
+      return true;
+    }
+
+    var revision = DeploymentsService.getRevision(replicaSet);
+
+    // If not part of a deployment, always show the replica set.
+    if (!revision) {
+      return true;
+    }
+
+    // If the deployment has been deleted and the replica set has no replicas, hide it.
+    // Otherwise all old replica sets for a deleted deployment will be visible.
+    if (!deployment) {
+      return false;
+    }
+
+    // Show the replica set if it's the latest revision.
+    return DeploymentsService.getRevision(deployment) === revision;
+  };
+
+  // Group replica sets by deployment and filter the visible replica sets.
+  var groupReplicaSets = function() {
+    if (!overview.replicaSets || !overview.deployments) {
+      return;
+    }
+
+    overview.replicaSetsByDeployment = LabelsService.groupBySelector(overview.replicaSets, overview.deployments, { matchSelector: true });
+    overview.currentByDeployment = {};
+
+    // Sort the visible replica sets.
+    _.each(overview.replicaSetsByDeployment, function(replicaSets, deploymentName) {
+      if (!deploymentName) {
+        return;
+      }
+
+      var deployment = overview.deployments[deploymentName];
+      var visibleRelicaSets = _.filter(replicaSets, function(replicaSet) {
+        return isReplicaSetVisible(replicaSet, deployment);
+      });
+      var ordered = DeploymentsService.sortByRevision(visibleRelicaSets);
+      overview.replicaSetsByDeployment[deploymentName] = ordered;
+      // TODO: Need to check if this really works for failed / canceled rollouts.
+      // It might need to be reworked.
+      overview.currentByDeployment[deploymentName] = _.head(ordered);
+      // var deploymentRevision = DeploymentsService.getRevision(deployment);
+      // overview.currentByDeployment[deploymentName] = _.find(replicaSets, function(replicaSet) {
+      //   return DeploymentsService.getRevision(replicaSet) === deploymentRevision;
+      // });
+    });
+    overview.vanillaReplicaSets = _.sortBy(overview.replicaSetsByDeployment[''], 'metadata.name');
+
+    // Since the visible replica sets for each deployment have changed, update
+    // the deployment warnings.
+    updateAllDeploymentWarnings();
+  };
+
+  // Find the services that direct traffic to each API object.
+  //
+  // Updates `state.servicesByObjectUID`
+  //   key: object UID
+  //   value: array of sorted services
+  var selectorsByService = {};
+  var updateServicesForObjects = function(apiObjects) {
+    if (!apiObjects || !overview.services) {
+      return;
+    }
+
+    _.each(apiObjects, function(apiObject) {
+      var services = [];
+      var uid = getUID(apiObject);
+      var podTemplate = getPodTemplate(apiObject) || { metadata: { labels: {} } };
+      _.each(selectorsByService, function(selector, serviceName) {
+        if (selector.matches(podTemplate)) {
```

Comment:
It looks like `LabelSelector.matches` handles null and undefined values. I'll remove the dummy object.

https://github.com/kubernetes-ui/label-selector/blob/54cc2dba5a2229073088f6dee589981ace485e6d/labelSelector.js#L98-L101